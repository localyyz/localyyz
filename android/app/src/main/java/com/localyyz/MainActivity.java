package com.localyyz;

import android.content.Intent;
import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;

import javax.annotation.Nullable;

import io.branch.rnbranch.RNBranchModule;

public class MainActivity extends ReactActivity {

    /**
     * Returns the name of the main component registered from JavaScript.
     * This is used to schedule rendering of the component.
     */
    @Override
    protected String getMainComponentName() {
        return "localyyz";
    }

    @Override
    protected void onStart() {
        super.onStart();
        RNBranchModule.initSession(getIntent().getData(), this);
    }

    @Override
    public void onNewIntent(Intent intent) {
        setIntent(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MainApplication.getCallbackManager().onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Nullable
            @Override
            protected Bundle getLaunchOptions() {
                Bundle initialProps = new Bundle();
                initialProps.putString("API_URL", BuildConfig.API_URL);
                return initialProps;
            }
        };
    }

}
