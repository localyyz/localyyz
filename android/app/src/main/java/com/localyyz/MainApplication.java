package com.localyyz;

import android.app.Application;

import com.BV.LinearGradient.LinearGradientPackage;
import com.cmcewen.blurview.BlurViewPackage;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.ReactApplication;
import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
import com.idehub.GoogleAnalyticsBridge.GoogleAnalyticsBridgePackage;
import com.brentvatne.react.ReactVideoPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.facebook.soloader.SoLoader;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.microsoft.codepush.react.CodePush;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnative.photoview.PhotoViewPackage;

import java.util.Arrays;
import java.util.List;

import ca.jaysoo.extradimensions.ExtraDimensionsPackage;
import io.branch.referral.Branch;
import io.branch.rnbranch.RNBranchPackage;

//import com.reactnativepayments.ReactNativePaymentsPackage;

public class MainApplication extends Application implements ReactApplication {

  private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }
    
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new ReactNativeOneSignalPackage(),
            new GoogleAnalyticsBridgePackage(),
            new ReactVideoPackage(),
            new ExtraDimensionsPackage(),
              new CodePush(BuildConfig.CODEPUSH_KEY, getApplicationContext(), BuildConfig.DEBUG),
              new RNBranchPackage(),
              //new ReactNativePaymentsPackage(),
              new FBSDKPackage(mCallbackManager),
              new PhotoViewPackage(),
              new VectorIconsPackage(),
              new LinearGradientPackage(),
              new RNDeviceInfo(),
              new BlurViewPackage()
      );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
      return getPackages();
    }


    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    FacebookSdk.sdkInitialize(getApplicationContext());
    // If you want to use AppEventsLogger to log events.
    AppEventsLogger.activateApp(this);

    // Branch
    Branch.getAutoInstance(this);

    SoLoader.init(this, /* native exopackage */ false);
  }

}
