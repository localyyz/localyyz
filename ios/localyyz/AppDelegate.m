/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

//#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

// Branch
#import <react-native-branch/RNBranch.h>

// Facebook
#import <FBSDKCoreKit/FBSDKCoreKit.h>
#import <FBSDKLoginKit/FBSDKLoginKit.h>
#import <FBNotifications/FBNotifications.h>
#import <UserNotifications/UserNotifications.h>

// CodePush
#import <CodePush/CodePush.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  if (RCT_DEBUG == 1) {
    //jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
    jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios&dev=true"];
  } else {
    jsCodeLocation = [CodePush bundleURL];
  }

  NSString *apiUrl = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"API_URL"];
  NSDictionary *props = @{@"API_URL" : apiUrl};
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"localyyz"
                                               initialProperties:props
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

  // replace RN load view with the launch screen
  UIView* launchScreenView = [[[NSBundle mainBundle] loadNibNamed:@"LaunchScreen" owner:self options:nil] objectAtIndex:0];
  launchScreenView.frame = self.window.bounds;
  rootView.loadingView = launchScreenView;

  // facebook
  [[FBSDKApplicationDelegate sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];

  // branch
  [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];

  // push notification
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
  [center requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error) {
    if( !error ) {
      // required to get the app to do anything at all about push notifications
      [[UIApplication sharedApplication] registerForRemoteNotifications];
      NSLog( @"Push registration success.");
    } else {
      NSLog( @"Push registration FAILED" );
      NSLog( @"ERROR: %@ - %@", error.localizedFailureReason, error.localizedDescription );
      NSLog( @"SUGGESTIONS: %@ - %@", error.localizedRecoveryOptions, error.localizedRecoverySuggestion );
    }
  }];

  return YES;
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler {
  FBNotificationsManager *notificationsManager = [FBNotificationsManager sharedManager];
  [notificationsManager presentPushCardForRemoteNotificationPayload:userInfo
                                                 fromViewController:nil
                                                         completion:^(FBNCardViewController * _Nullable viewController, NSError * _Nullable error) {
                                                           if (error) {
                                                             completionHandler(UIBackgroundFetchResultFailed);
                                                           } else {
                                                             completionHandler(UIBackgroundFetchResultNewData);
                                                           }
                                                         }];
}

- (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  
  BOOL handled = [[FBSDKApplicationDelegate sharedInstance] application:application
                                                                openURL:url
                                                      sourceApplication:options[UIApplicationOpenURLOptionsSourceApplicationKey]
                                                             annotation:options[UIApplicationOpenURLOptionsAnnotationKey]
                  ];
  // Add any custom logic here.
  return handled;
}

// track opening and resuming the application attached to the push notification
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
  [FBSDKAppEvents logPushNotificationOpen:userInfo];
}

// track action performed that was attached to the push notification
- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo
  completionHandler:(void (^)())completionHandler {
  [FBSDKAppEvents logPushNotificationOpen:userInfo action:identifier];
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
  return [RNBranch continueUserActivity:userActivity];
}

- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  NSString *token = [[deviceToken description] stringByTrimmingCharactersInSet: [NSCharacterSet characterSetWithCharactersInString:@"<>"]];
  token = [token stringByReplacingOccurrencesOfString:@" " withString:@""];
  NSLog(@"content--- %@", token);
  [FBSDKAppEvents setPushNotificationsDeviceToken:deviceToken];
}

// Facebook SDK
- (void)applicationDidBecomeActive:(UIApplication *)application {
  [FBSDKAppEvents activateApp];
}

@end
