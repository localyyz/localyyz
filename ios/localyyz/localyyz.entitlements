<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
