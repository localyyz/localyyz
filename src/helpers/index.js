/*
 * @flow
 * @providesModule localyyz/helpers
 */

import { NavigationActions } from "react-navigation";
import { Alert } from "react-native";
import { observable, computed } from "mobx";
import getSymbolFromCurrency from "currency-symbol-map";
export { default as isCssColor } from "./csscolor";
export { default as linkParser } from "./linkparser";

// usage:
// - instead of the typical `@observable x;`
// pattern used by mobx, wrap the prop by
// `@box x;`
// it will now inherit the custom getter and setter functions
export const box = (target, name, descriptor) => {
  const privateName = `_${name}`;
  observable(target, privateName, descriptor);
  return computed(target, name, {
    get() {
      return this[privateName];
    },
    set(value) {
      this[privateName] = value;
    }
  });
};

export const resetHome = params => {
  let opt = {
    index: 0,
    actions: [
      NavigationActions.navigate({
        routeName: "App",
        params: params
      })
    ],
    key: null
  };
  return NavigationActions.reset(opt);
};

export const changeTab = (tab, params) => {
  return {
    type: "tab",
    routeName: tab,
    params: params
  };
};

export const resetAction = (routeName, params = {}) => {
  let opt = {
    index: 0,
    actions: [
      NavigationActions.navigate({
        routeName: routeName,
        params: params
      })
    ]
  };
  if (routeName === "Login") {
    opt.key = null;
  }
  return NavigationActions.reset(opt);
};

export const paramsAction = (key, params = {}) => {
  return NavigationActions.setParams({
    params: params,
    key: key
  });
};

export function randInt(max) {
  return Math.floor(Math.random() * Math.floor(max));
}

export function capitalize(string) {
  return (string + "").replace(/^(.)|\s+(.)/g, function($1) {
    return $1.toUpperCase();
  });
}

export function onlyIfLoggedIn({ hasSession }, action, navigation) {
  if (hasSession) {
    action && action();

    return true;
  } else if (navigation) {
    // only show CTA if navigation is given
    Alert.alert("Please login or create a new account", null, [
      {
        text: "Cancel",
        style: "cancel"
      },
      {
        text: "OK",
        onPress: () =>
          navigation.navigate("Login", {
            appContext: "product"
          })
      }
    ]);

    // used to trigger external events
    return false;
  }
}

export function toPriceString(price, currency = "USD", avoidFree = false) {
  return price != null && (price > 0 || !avoidFree)
    ? price > 0
      ? `${getSymbolFromCurrency(currency) || "$"}${price.toFixed(2)}`
      : "Free"
    : "";
}
