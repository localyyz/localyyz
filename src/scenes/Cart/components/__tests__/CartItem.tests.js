import React from "react";

// custom
import { renderWithLayout, findById } from "localyyz/tests";
import { toPriceString } from "localyyz/helpers";

// local
import ComponentWrapper from "../CartItem";
const Component = ComponentWrapper.wrappedComponent || ComponentWrapper;

const DEFAULT_PROPS = {
  item: {
    product: {
      associatedPhotos: [{ imageUrl: "picture.gif", width: 100, height: 100 }],
      title: "Roshe One",
      brand: "Nike",
      price: 100,
      previousPrice: 120
    },
    hasError: false,
    variant: {
      description: "Some shoes"
    },
    price: 100
  },
  removeItem: jest.fn(),
  navigation: {}
};

describe(Component.name, () => {
  it("should render properly to snapshot", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let [instance, layout] = [c.getInstance(), c.toJSON()];

    expect(instance).toBeDefined();
    expect(layout).toMatchSnapshot();
  });

  it("should render price properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "price").children[0]).toBe(
      toPriceString(DEFAULT_PROPS.item.price)
    );
  });

  it("should render prev price properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "previousPrice").children[0]).toBe(
      toPriceString(DEFAULT_PROPS.item.product.previousPrice)
    );
  });

  it("should render title properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "title").children[0]).toBe(
      DEFAULT_PROPS.item.product.title
    );
  });

  it("should render brand properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "brand").children[0]).toBe(
      DEFAULT_PROPS.item.product.brand
    );
  });

  it("should not show error", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "hasError")).not.toBeDefined();
  });

  it("should render photo properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "photo").props.source.uri).toBe(
      DEFAULT_PROPS.item.product.associatedPhotos[0].imageUrl
    );
  });

  it("should render variant info properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "variant").children[0]).toBe(
      DEFAULT_PROPS.item.variant.description
    );
  });

  it("should not show previous price if not on sale", () => {
    let MODIFIED_PROPS = DEFAULT_PROPS;
    MODIFIED_PROPS.item.product.previousPrice = 0;

    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "previousPrice")).not.toBeDefined();
  });

  it("should show OOS if hasError", () => {
    let MODIFIED_PROPS = DEFAULT_PROPS;
    MODIFIED_PROPS.item.hasError = true;

    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "hasError")).toBeDefined();
  });
});
