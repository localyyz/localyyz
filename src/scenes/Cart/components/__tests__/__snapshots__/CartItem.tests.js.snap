// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render properly to snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#dbddde",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {},
          Object {
            "left": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              Object {
                "width": 150,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "height": 100,
                  "overflow": "hidden",
                  "width": 100,
                }
              }
            >
              <Image
                constrainWidth={150}
                onError={[Function]}
                resizeMode="contain"
                shouldPinHeight={false}
                shouldPinWidth={false}
                source={
                  Object {
                    "uri": "picture.gif",
                  }
                }
                sourceHeight={100}
                sourceWidth={100}
                style={
                  Array [
                    Object {
                      "height": 100,
                      "width": 100,
                    },
                  ]
                }
                testID="photo"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "padding": 15,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "#000000",
                      "fontSize": 16,
                      "fontWeight": "600",
                    }
                  }
                  testID="title"
                >
                  Roshe One
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "#000000",
                      "fontSize": 16,
                      "fontWeight": "600",
                    },
                    Object {
                      "marginLeft": 15,
                    },
                  ]
                }
                testID="price"
              >
                $100
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    }
                  }
                  testID="brand"
                >
                  Nike
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    },
                    Object {
                      "marginLeft": 15,
                    },
                    Object {
                      "textDecorationLine": "line-through",
                    },
                  ]
                }
                testID="previousPrice"
              >
                $120
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  Object {
                    "marginTop": 7.5,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    }
                  }
                  testID="variant"
                >
                  Some shoes
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View />
    <View />
  </View>
</View>
`;
